lib/std/meta.zig

```diff
@@ -979,9 +979,59 @@ test "std.meta.cast" {
 /// Given a value returns its size as C's sizeof operator would.
 /// This is for translate-c and is not intended for general use.
 pub fn sizeof(target: anytype) usize {
-    switch (@typeInfo(@TypeOf(target))) {
-        .Type => return @sizeOf(target),
-        .Float, .Int, .Struct, .Union, .Enum => return @sizeOf(@TypeOf(target)),
+    const T: type = if (@TypeOf(target) == type) target else @TypeOf(target);
+    switch (@typeInfo(T)) {
+        .Float, .Int, .Struct, .Union, .Enum, .Array, .Bool, .Vector => return @sizeOf(T),
+        .Fn => {
```

Comment:
```suggestion
        .Fn, .BoundFn => {
```

Reply:
Why shouldn't bound functions be an error?

Reply:
Translate C will not generate bound functions, this is not needed.

Reply:
Why do you limit the use of this function only for translate C?

Reply:
Because that's what it was before I made these changes.  I am simply trying to unblock someone who was running into the TODO compile error at the bottom.